src/mp2_ri_grad_util.F

```diff
@@ -1665,17 +1668,23 @@ END SUBROUTINE create_dbcsr_gamma
 ! **************************************************************************************************
 !> \brief ...
 !> \param mat ...
+!> \param unitout ...
 ! **************************************************************************************************
-   SUBROUTINE write_array(mat)
+   SUBROUTINE write_array(mat, unitout)
       REAL(KIND=dp), DIMENSION(:, :), INTENT(IN)         :: mat
+      INTEGER, INTENT(IN), OPTIONAL                      :: unitout
 
       INTEGER                                            :: iii, jjj
 
       WRITE (*, *)
       DO iii = 1, SIZE(mat, 1)
          WRITE (*, *) iii
          DO jjj = 1, SIZE(mat, 2), 20
-            WRITE (*, '(1000F10.5)') mat(iii, jjj:MIN(SIZE(mat, 2), jjj+19))
+            IF (PRESENT(unitout)) THEN
+               WRITE (1000+unitout, '(1000F10.5)') mat(iii, jjj:MIN(SIZE(mat, 2), jjj+19))
```

Comment:
is it intentional that only the inner loop output goes to `1000+unitout` and the empty lines and outer index still to the default?

Reply:
I don't really know. We should ask that the author of these lines (probably Mauro). The routine was part of the debug code. So, I assume it was not meant for production but for debugging purposes only.